docs/FATE_On_Spark_With_Pulsar.md

```diff
@@ -0,0 +1,338 @@
+
+The architecture is shown in the following diagram. The Pulsar service "10000.fate.org" belongs to the organization with ID 10000, and the pulsar service "9999.fate.org" belongs to the organization with ID 9999, and "proxy.fate.org" is ats Service is the center of the star network.
+
+<div style="text-align:center", align=center>
+<img src="./images/pulsar_sni_proxy.png" />
+</div>
+
+#### Certificate Generation
+
+Since the SNI proxy is based on TLS, it is necessary to configure the certificate for the ATS and Pulsar services.
+
+##### Generate CA's Certificate
+
+Enter the following command to create a directory for the CA, and put the openssl configuration file in the directory.
+
+``` bash
+$ mkdir my-ca
+$ cd my-ca
+$ wget https://raw.githubusercontent.com/apache/pulsar/master/site2/website/static/examples/openssl.cnf
+$ export CA_HOME=$(pwd)
+```
+
+Enter the following commands to create the necessary directories, keys, and certificates.
+
+``` bash
+$ mkdir certs crl newcerts private
+$ chmod 700 private/
+$ openssl genrsa -aes256 -out private/ca.key.pem 4096
+$ touch index.txt
+$ echo 1000 > serial
+$ chmod 400 private/ca.key.pem
+$ openssl req -config openssl.cnf -key private/ca.key.pem \
+    -new -x509 -days 7300 -sha256 -extensions v3_ca \
+    -out certs/ca.cert.pem
+$ chmod 444 certs/ca.cert.pem
+```
+
+Beyond the above command, further information is required to generate the key and certificate. Once the above command is completed, the CA-related certificates and keys have been generated, they are:
+- certs/ca.cert.pem   the certification of CA
+- private/ca.key.pem  the key of CA
+
+##### Generate a certificate for "10000.fate.org"
+
+1. Generate the directory
+```
+$ mkdir 10000.fate.org
+```
+
+2. Generate private key
+```
+$ openssl genrsa -out 10000.fate.org/broker.key.pem 2048
+```
+
+3. Transfer format
+```
+$ openssl pkcs8 -topk8 -inform PEM -outform PEM \
+      -in 10000.fate.org/broker.key.pem -out 10000.fate.org/broker.key-pk8.pem -nocrypt
+```
+
+4. Enter the following command to generate a certificate request, where the `Common Name` is set to "10000.fate.org"
+```
+$ openssl req -config openssl.cnf \
+    -key 10000.fate.org/broker.key.pem -new -sha256 -out 10000.fate.org/broker.csr.pem
+```
+
+5. Enter the following command to sign the certificate with CA's private key
+```
+$ openssl ca -config openssl.cnf -extensions server_cert \
+    -days 1000 -notext -md sha256 \
+    -in 10000.fate.org/broker.csr.pem -out 10000.fate.org/broker.cert.pem
+```
+
+At this time, the certificate "broker.cert.pem" and a key "broker.key-pk8.pem" are stored in the "10000.fate.org" directory.
+
+##### Generate a certificate for "9999.fate.org"
+
+The generation of the "9999.fate.org" certificate is consistent with the above steps, however, the Common Name step 4 is "9999.fate.org".
+
+The following operation will assume that the certificate of "9999.fate.org" has been generated.
```

Comment:
The generation of the "9999.fate.org" certificate is consistent with the above steps, however, the Common Name step 4 is "9999.fate.org". ? Not sure 
The following operation will assume -> The following operations assume